shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -877,13 +913,45 @@ SemanticsObject hitTest(float[] point) {
         boolean isFocusable() {
             int scrollableActions = Action.SCROLL_RIGHT.value | Action.SCROLL_LEFT.value
                     | Action.SCROLL_UP.value | Action.SCROLL_DOWN.value;
+            if (hasFlag(Flag.IS_EDGE) && value == null) {
+                return false;
+            }
             return (actions & ~scrollableActions) != 0
                 || flags != 0
                 || (label != null && !label.isEmpty())
                 || (value != null && !value.isEmpty())
                 || (hint != null && !hint.isEmpty());
         }
 
+        void collectEdges(List<SemanticsObject> edges) {
+            if (hasFlag(Flag.IS_EDGE)) {
+                edges.add(this);
+            }
+            if (children != null) {
+                for (int i = 0; i < children.size(); ++i) {
+                    children.get(i).collectEdges(edges);
+                }
+            }
+        }
+
+        String getRouteName() {
+            String name = null;
+            if (children != null) {
+                for (int i = children.size() - 1; i >= 0; --i) {
+                    String newName = children.get(i).getRouteName();
+                    if (newName != null && !newName.isEmpty()) {
+                        name = newName;
+                    }
+                }
+            }
+            if (hasFlag(Flag.IS_ROUTE)) {
+                if (value != null && !value.isEmpty()) {
+                    name = value;
```

Comment:
If you get here, all the expensive work in the loop above is thrown away. Should we be checking this first?

I don't really understand the contract you're trying to implement here. Might be worth a comment.

Reply:
D'oh